packages/core/src/services/evaluationResults/aggregations/modalValueQuery.ts

```diff
@@ -0,0 +1,87 @@
+import { and, eq, sql, sum } from 'drizzle-orm'
+
+import { getCommitFilter } from '../_createEvaluationResultQuery'
+import { Commit, Evaluation } from '../../../browser'
+import { database } from '../../../client'
+import { EvaluationResultsRepository } from '../../../repositories'
+import { DocumentLogsRepository } from '../../../repositories/documentLogsRepository'
+import { commits } from '../../../schema'
+
+export async function getEvaluationModalValueQuery(
+  {
+    workspaceId,
+    evaluation,
+    documentUuid,
+    commit,
+  }: {
+    workspaceId: number
+    evaluation: Evaluation
+    documentUuid: string
+    commit: Commit
+  },
+  db = database,
+) {
+  const { scope: evaluationResultsScope } = new EvaluationResultsRepository(
+    workspaceId,
+    db,
+  )
+  const { scope: documentLogsScope } = new DocumentLogsRepository(
+    workspaceId,
+    db,
+  )
+
+  const totalQuery = await db
+    .select({
+      totalCount: sum(evaluationResultsScope.id)
+        .mapWith(Number)
+        .as('ev_results_from_commit_total'),
+    })
+    .from(evaluationResultsScope)
+    .innerJoin(
+      documentLogsScope,
+      eq(documentLogsScope.id, evaluationResultsScope.documentLogId),
+    )
+    .innerJoin(commits, eq(commits.id, documentLogsScope.commitId))
+    .where(
+      and(
+        eq(evaluationResultsScope.evaluationId, evaluation.id),
+        eq(documentLogsScope.documentUuid, documentUuid),
+        getCommitFilter(commit),
+      ),
+    )
+    .groupBy(evaluationResultsScope.evaluationId)
+    .limit(1)
+
+  const mostCommonResult = await db
+    .select({
+      mostCommon: evaluationResultsScope.result,
+      mostCommonCount: sum(evaluationResultsScope.id)
+        .mapWith(Number)
+        .as('most_common_count'),
+    })
+    .from(evaluationResultsScope)
+    .innerJoin(
+      documentLogsScope,
+      eq(documentLogsScope.id, evaluationResultsScope.documentLogId),
+    )
+    .innerJoin(commits, eq(commits.id, documentLogsScope.commitId))
+    .where(
+      and(
+        eq(evaluationResultsScope.evaluationId, evaluation.id),
+        eq(documentLogsScope.documentUuid, documentUuid),
+        getCommitFilter(commit),
+      ),
+    )
+    .groupBy(evaluationResultsScope.result)
+    .orderBy(sql`most_common_count DESC`)
+    .limit(1)
+
+  const total = totalQuery[0]!.totalCount
+  const mostCommon = mostCommonResult[0]!.mostCommon
+  const mostCommonCount = mostCommonResult[0]!.mostCommonCount
```

Comment:
shouldn't we add some protections instead of assuming they will be there? keep in mind ! does **not** mean typescript will throw an error if the value is indeed null or undefined

Reply:
It will, won't it? It would rais a `Can't read attribute 'mostCommonCount' from 'undefined'` or something like that

Reply:
yes but that's an unexpected javascript error, what i mean is that typescript ! operator does not throw a fancy "you said this value will be there and it's not there" kind of error

Reply:
Fixed